### DISCUSSION TITLE
Postgres CREATE OR REPLACE FUNCTION

### DISCUSSION THREAD

**Discussion by solisgb** [dbeaver/dbeaver#15122] [Q&A]
#### System information: 
- Operating system (distribution) and version: Windows 10 v. 20H2
- DBeaver version 21.3.2.202112261636
- Additional extensions: Postgis

#### Connection specification:
- Database name and version : PostgreSQL 12.8, compiled by Visual C++ build 1914, 64-bit
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 

#### Describe the problem you're observing:
When I type the code below in the SQL console and run it I get the following error: SQL Error [42601]: ERROR: error de sintaxis en o cerca de «return» (sintax error near return)
  Position: 1

This is te code:
create or replace function tmp(x real)
returns real
as
$$
declare
	x1 real := 1.0;
    xsum real := 0.0;
begin

	if x  > 0 then
		xsum = xsum + x1;
	else
		xsum = xsum - x1;
	end if;
   
	return xsum;
  
end;
$$ language plpgsql;

If I run the same code in pdAdmin no error occurs: Query returned successfully in 74 msec.  

The function saved through pdAdmin is:
CREATE OR REPLACE FUNCTION public.tmp(IN x real)
    RETURNS real
    LANGUAGE 'plpgsql'
    VOLATILE
    PARALLEL UNSAFE
    COST 100
    
AS $BODY$
declare
	x1 real := 1.0;
    xsum real := 0.0;
begin

	if x  > 0 then
		xsum = xsum + x1;
	else
		xsum = xsum - x1;
	end if;
   
	return xsum;
  
end;
$BODY$;

If I run exactly the code saved in PgAdmin in DBeaver I get the error: SQL Error [42601]: ERROR: error de sintaxis en o cerca de «return» (sintax error near return)
  Position: 1

The code successfully executed from PgAdmin differs slightly from the view provided by PgAdmin (no lines: VOLATILE; PARALLEL UNSAFE; COST 100). The view in DBeaver is:
CREATE OR REPLACE FUNCTION public.tmp(x real)
 RETURNS real
 LANGUAGE plpgsql
AS $function$
declare
	x1 real := 1.0;
    xsum real := 0.0;
begin

	if x  > 0 then
		xsum = xsum + x1;
	else
		xsum = xsum - x1;
	end if;
   
	return xsum;
  
end;
$function$
;

**Comment by LonwoLonwo** [comment]
Hello @solisgb 

I've tried to execute your statement in DBeaver SQL Editor, and the statement works fine for me. With highlight and Ctrl+Enter.

![2022-01-04 23_31_18-DBeaver Ultimate 21 3 2 - _postgres 13 3_Script-81](https://user-images.githubusercontent.com/45152336/148120481-3537dfcb-7c01-46fd-bee5-b62bc26ab68e.png)

Please can you check your preferences? Window -> Preferences -> Editors -> SQL Editors -> SQl Processing. 
If the "Blank line is statement delimiter" setting is enabled, you can disable this setting, or you can remove blank lines from your statement. It helped me to execute your statement at the last delimiter without highlighting, so it must help you.

![2022-01-04 23_36_25-DBeaver Ultimate 21 3 2 - _postgres 13 3_Script-81](https://user-images.githubusercontent.com/45152336/148121393-e72b444f-a105-425b-a478-c5598f616e55.png)

**Comment by solisgb** [comment]
Thank you very much,
Following your indications my problem has been solved.

El mar, 4 ene 2022 a las 21:42, Anastasiya ***@***.***>)
escribió:

> Hello @solisgb <https://github.com/solisgb>
>
> I've tried to execute your statement in DBeaver SQL Editor, and the
> statement works fine for me. With highlight and Ctrl+Enter.
>
> [image: 2022-01-04 23_31_18-DBeaver Ultimate 21 3 2 - _postgres 13
> 3_Script-81]
> <https://user-images.githubusercontent.com/45152336/148120481-3537dfcb-7c01-46fd-bee5-b62bc26ab68e.png>
>
> Please can you check your preferences? Window -> Preferences -> Editors ->
> SQL Editors -> SQl Processing.
> If the "Blank line is statement delimiter" setting is enabled, you can
> disable this setting, or you can remove blank lines from your statement. It
> helped me to execute your statement at the last delimiter without
> highlighting, so it must help you.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dbeaver/dbeaver/issues/15111#issuecomment-1005155701>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJBSDBZHD2W5R3TBM4VVZFLUUNLTBANCNFSM5LHKWYEA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Luis Solís García-Barbón*
Si usted recibe este mensaje por error, por favor, comuníquelo al remitente
y borre el mensaje y los archivos adjuntos de su sistema.
If you receive this message by mistake, please inform the sender and delete
the message and attachments from your system.

*No me imprimas si no es necesario*

*Don't print me unless it's necessary*
